layers/gpu/spirv/non_bindless_oob_texel_buffer_pass.cpp

```diff
@@ -0,0 +1,234 @@
+                                                           const InjectionData& injection_data) {
+    assert(access_chain_inst_ && var_inst_);
+    const Constant& set_constant = module_.type_manager_.GetConstantUInt32(descriptor_set_);
+    const Constant& binding_constant = module_.type_manager_.GetConstantUInt32(descriptor_binding_);
+    const uint32_t descriptor_index_id = CastToUint32(descriptor_index_id_, block, inst_it);  // might be int32
+
+    const uint32_t opcode = target_instruction_->Opcode();
+    const uint32_t image_operand_position = OpcodeImageOperandsPosition(opcode);
+    if (target_instruction_->Length() > image_operand_position) {
+        const uint32_t image_operand_word = target_instruction_->Word(image_operand_position);
+        if ((image_operand_word & (spv::ImageOperandsConstOffsetMask | spv::ImageOperandsOffsetMask)) != 0) {
+            // TODO - Add support if there are image operands (like offset)
+        }
+    }
+
+    // Use the imageFetch() parameter to decide the offset
+    // TODO - This assumes no depth/arrayed/ms from AnalyzeInstruction
+    descriptor_offset_id_ = CastToUint32(target_instruction_->Operand(1), block, inst_it);
+
+    const uint32_t function_result = module_.TakeNextId();
+    const uint32_t function_def = GetLinkFunctionId();
+    const uint32_t bool_type = module_.type_manager_.GetTypeBool().Id();
+
+    block.CreateInstruction(
+        spv::OpFunctionCall,
+        {bool_type, function_result, function_def, injection_data.inst_position_id, injection_data.stage_info_id,
+         descriptor_array_size_id_, set_constant.Id(), binding_constant.Id(), descriptor_index_id, descriptor_offset_id_},
+        inst_it);
+
+    return function_result;
+}
+
+void NonBindlessOOBTexelBufferPass::Reset() {
+    access_chain_inst_ = nullptr;
+    var_inst_ = nullptr;
+    target_instruction_ = nullptr;
+    descriptor_array_size_id_ = 0;
+    descriptor_set_ = 0;
+    descriptor_binding_ = 0;
+    descriptor_index_id_ = 0;
+    descriptor_offset_id_ = 0;
+}
+
+bool NonBindlessOOBTexelBufferPass::AnalyzeInstruction(const Function& function, const Instruction& inst) {
+    const uint32_t opcode = inst.Opcode();
+
+    if (opcode != spv::OpImageFetch && opcode != spv::OpImageWrite && opcode != spv::OpImageRead) {
+        return false;
+    }
+    const uint32_t image_word = OpcodeImageAccessPosition(opcode);
+
+    image_inst_ = function.FindInstruction(inst.Word(image_word));
+    if (!image_inst_) return false;
+    const Type* image_type = module_.type_manager_.FindTypeById(image_inst_->TypeId());
+    if (!image_type) return false;
+
+    const uint32_t dim = image_type->inst_.Operand(1);
+    if (dim != spv::DimBuffer) {
+        return false;  // It is a Storage Image
+    }
+    const uint32_t depth = image_type->inst_.Operand(2);
+    const uint32_t arrayed = image_type->inst_.Operand(3);
+    const uint32_t ms = image_type->inst_.Operand(4);
```

Comment:
```suggestion
    const uint32_t multi_sampling = image_type->inst_.Operand(4);
```
if it is indeed multi sampling?